### PR TITLE
Fix (almost) all vulkan validation errors.

### PR DESCRIPTION
**Pull request by DadSchoorse**
The only ones left are related to the mesa specific pNext chain struct.

### CODE REVIEW

File: src/rendervulkan.cpp [hunk 1]
```diff
@@ -1443,8 +1444,14 @@ void fini_device()
 
 bool acquire_next_image( void )
 {
-	VkResult res = vkAcquireNextImageKHR( device, g_output.swapChain, UINT64_MAX, VK_NULL_HANDLE, VK_NULL_HANDLE, &g_output.nSwapChainImageIndex );
-	return res == VK_SUCCESS || res == VK_SUBOPTIMAL_KHR;
+	VkResult res = vkAcquireNextImageKHR( device, g_output.swapChain, UINT64_MAX, VK_NULL_HANDLE, g_output.acquireFence, &g_output.nSwapChainImageIndex );
+	if ( res != VK_SUCCESS && res != VK_SUBOPTIMAL_KHR )
+		return false;
+	res = vkWaitForFences( device, 1, &g_output.acquireFence, false, UINT64_MAX );
+	if ( res != VK_SUCCESS )
```
Comment (Joshua-Ashton) [inline src/rendervulkan.cpp:1456]: Just put the vkWaitForFences in the if

Comment (Joshua-Ashton) [reply src/rendervulkan.cpp:1456]: Same for ResetFences

File: src/rendervulkan.cpp [hunk 2]
```diff
@@ -1102,6 +1102,11 @@ static bool init_device()
 	clockFeatures.pNext = (void *) std::exchange(deviceCreateInfo.pNext, &clockFeatures);
 	clockFeatures.shaderDeviceClock = VK_TRUE;
 
+	VkPhysicalDeviceSamplerYcbcrConversionFeatures ycbcrFeatures = {};
+	ycbcrFeatures.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SAMPLER_YCBCR_CONVERSION_FEATURES;
+	ycbcrFeatures.pNext = (void *) std::exchange(deviceCreateInfo.pNext, &ycbcrFeatures);
```
Comment (Joshua-Ashton) [inline src/rendervulkan.cpp:1098]: Probably makes sense to std exchange this onto device features if we have it to avoid the unnecessary cast?

**Comment by Joshua-Ashton** [comment]
Thanks. I merged this manually except for ee122e92853e6931ed96fc89c90a1e60168f549d

**Comment by DadSchoorse** [comment]
What's the issue with ee122e92853e6931ed96fc89c90a1e60168f549d? It should be more correct than what we do rn.

**Pull request closed without merging**